optaplanner-examples/src/main/java/org/optaplanner/examples/machinereassignment/solver/score/MrConstraintName.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.machinereassignment.solver.score;
+
+enum MrConstraintName {
```

Comment:
Intresting that you're using an enum. It crossed my mind too.

For other examples, we've always just use public static strings - it should be consistent (either we refactor them too or we refactor this).
The problem with enums is that the Solver API expects a String constraintName, not an enum - so you'd need to translate every string coming out of the Indictement/Match api first - which is overhead to just keeping it as a String. Wdyt?

Reply:
Is there a good reason for an interface with constants these days? It comes from the time before enums, AFAIK.
There is no problem with keeping the Solver API as is; one just needs to pass there the enum's string representation. Maybe I don't fully understand the issue.
As to the consistency, I am definitely not turning all examples to enums as a part of this PR to your fork of OptaPlanner :) let's do it in later in a PR that goes to optaplanner/master.

Reply:
Do you insist on turning this into an interface with constants and we rewrite everything later? For me, having an enum here and constants everywhere else is not a big issue, especially if that's just a temporary state.